src/test/java/reactor/kafka/receiver/internals/MockReceiverTest.java

```diff
@@ -596,15 +596,17 @@ public void manualAckCommitBatchSize() {
         AtomicLong lastCommitted = new AtomicLong(-1);
         sendMessages(topic, 0, 20);
         receiveAndVerify(15, r -> {
-            long offset = r.receiverOffset().offset();
-            if (offset < 10) {
-                r.receiverOffset().acknowledge();
-                if (((offset + 1) % batchSize) == 0)
-                    lastCommitted.set(offset);
-            } else
-                uncommittedMessages.add(r);
-            verifyCommit(r, lastCommitted.get());
-            return Mono.just(r);
+            return Mono.fromCallable(() -> {
```

Comment:
I spent ~2 days debugging the tests after applying that small change.

"never block the emitter thread" makes a lot more sense now 😂